Makefile

```diff
@@ -126,6 +126,13 @@ OD			:=	${CROSS_COMPILE}objdump
 NM			:=	${CROSS_COMPILE}nm
 PP			:=	${CROSS_COMPILE}gcc -E
 
+# Use ${LD}.bfd instead if it exists (as absolute path or together with $PATH).
+ifneq ($(strip $(wildcard ${LD}.bfd) \
+	$(foreach dir,$(subst :, ,${PATH}),$(wildcard ${dir}/${LD}.bfd))),)
+LD			:=	${LD}.bfd
+TF_CFLAGS		+=	-fuse-ld=bfd
```

Comment:
@jwerner-chromium 
Apologies for the delay... this is breaking our clang / armclang builds. 

clang: error: argument unused during compilation: '-fuse-ld=bfd' [-Werror,-Wunused-command-line-argument]

I tested a noddy fix which was to wrap lines 129-134 in another ifneq that checks if CC is clang or armclang and that seemed to work. But not sure if this is the 'right' solution!

Build options being used:
clang: CROSS_COMPILE=aarch64-linux-gnu- V=1 DEBUG=1 PLAT=fvp CC=clang
armclang: CROSS_COMPILE=aarch64-linux-gnu- V=1 DEBUG=1 PLAT=fvp CC=<path/to/armclang/6.8/installation>/bin/armclang

Reply:
Ahh... hmm... if I understand this right, it actually understands the -fuse-ld option but it complains when it is passed in an invocation that doesn't really link the binary (i.e. with -c).

It looks like Trusted Firmware never actually calls $(CC) without -c, it always calls $(LD) directly for linking. So I think I can just remove the -fuse-ld part completely. Will update.